### PR TITLE
Use `cache: pip` from actions setup-python

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
the setup-python action now [offers package caching](https://github.com/actions/setup-python#caching-packages-dependencies).  It works well, hits a bit more often than our current cache, (and also works for tox deps, I checked)

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: .github/workflows/test_pull_requests.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-python@v3
+      - uses: actions/setup-python@v4
         with:
           python-version: "3.x"
```
Comment (Czaki) [inline .github/workflows/test_pull_requests.yml:20]: why no cache here?

File: .github/workflows/test_pull_requests.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ jobs:
...
```
Comment (Czaki) [inline .github/workflows/test_pull_requests.yml:35]: as above

**Review by Czaki — COMMENTED** [review]
Maybe also in test_typing, test_vendored, build_docs?

File: .github/workflows/test_pull_requests.yml (hunk 1 above)
Comment (tlambert03) [reply .github/workflows/test_pull_requests.yml:20]: just cause it's only two packages, it is never the rate limiting job.  keeping things simple if they don't need it

File: .github/workflows/test_pull_requests.yml (hunk 2 above)
Comment (tlambert03) [reply .github/workflows/test_pull_requests.yml:35]: same, overkill

**Comment by Carreau** [comment]
I originally did not use out of the box caches, as I believe pytorch was too big and made caching fail.
If that now works all good for me. Though at a cursory glance it seem to add about a minute or so to each linux test suite. Which I think in our current case is fine.

Just want to make sure we keep an eye on making sure the caching is not failing.

Do you know of we can ask tox to create env in a separate step so we can get separate timing for installation and actual test suite ?

**Review by Carreau — APPROVED** [review]

**Comment by tlambert03** [comment]
> Though at a cursory glance it seem to add about a minute or so to each linux test suite. 

yeah, even rerunning a few times, this seems to be the case.  In light of that, I'd probably just close it.

> Do you know of we can ask tox to create env in a separate step so we can get separate timing for installation and actual test suite ?

yeah, `tox --notest`

**Comment by Carreau** [comment]
> yeah, even rerunning a few times, this seems to be the case. In light of that, I'd probably just close it.
> 

Ok, up to you. I'm also happy to lose a minute or two, if we decrease bandwidth usage on PyPI.

> yeah, `tox --notest`

Thanks! I'll see if I can send a PR with just that, at least that will give us a better indicator with PR like this one if it has a strong effect on install time.

**Pull request merged**